honeybee_grasshopper_core/src/HB Aperture.py

```diff
@@ -0,0 +1,82 @@
+# Honeybee: A Plugin for Environmental Analysis (GPL)
+# This file is part of Honeybee.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Honeybee; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Create Honeybee Face
+-
+
+    Args:
+        _geo: Rhino Brep geometry.
+        _name_: A name for the Aperture. If the name is not provided a random
+            name will be assigned
+        operable_: Boolean to note whether the Aperture can be opened for ventilation.
+            Default: False.
+        ep_constr_: Optional text for the Aperture's energy construction to be looked
+            up in the construction library. This can also be a custom WindowConstruction
+            object. If no energy construction is input here, a default will be
+            assigned based on the properties of the parent face that the Aperture
+            is assigned to (ie. whether the Face is a RoofCeiling, whether it has
+            a Surface boundary condition, etc.)
+        rad_mat_: Optional text for the Aperture's radiance material to be looked
```

Comment:
My main point is about the approach itself. I think we should avoid name inputs as strings as much as possible. Adding one more component will make it more explicit and user can check that they getting the construction that they are expecting to get.

Reply:
updated my comment with longer explanation.

Reply:
I agree we need a longer discussion. I need to understand what "mistakes" you are talking about and how adding ~10 new components to get different types of materials, constructions, schedules, scheduletypelimits, constructionsets, programtypes, etc. from the library by name will help mitigate them.

I also realized that my second interpretation about duplicate code really doesn't make much sense since it's just a difference of 2 lines of code if we want to support retrieval of an item by name from the library.

Reply:
For the record, the "mistakes" that Mostapha was referring to were cases where people would overwrite materials or constructions in their library if they used the same name as something in their library. This will be addressed by keeping the library static from the perspective of Grasshopper. The only way that people will be able to edit their libraries is by dropping files into the folders of the library and, if people want to have custom materials generated within their Grasshopper scripts, they will have to be fully-detailed objects and they aren't going to get added to the library.